packages/icons/src/common/intent.ts

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the terms of the LICENSE file distributed with this project.
+ */
+
+/**
+ * The four basic intents.
+ */
+export type Intent = "none" | "primary" | "success" | "warning" | "danger";
```

Comment:
oh clever it's not an enum here, just a union of literals. i really want string literal support for our enum types like `Intent` and `Position` but this is not the time or place to do it. let's duplicate the `enum Intent` here so we don't end up with two different types with the same name.

Reply:
The problem is that a value of the duplicated `enum Intent` type cannot be assigned to using `enum Intent` from `core` (different packages, Typescript doesn't notice/care that the "shape" is identical). `enum Intent` *is* assignable to a string union though. I should have added a comment about this being a hack. Should we drop the union anyway?